src/libraries/Microsoft.PowerFx.Core/Public/Values/FormulaValue.cs

```diff
@@ -88,6 +89,68 @@ public bool TryGetPrimitiveValue(out object val)
 
             val = null;
             return false;
+        }
+
+        public bool AsBoolean()
+        {
+            if (this is BlankValue)
+            {
+                return default;
+            }
+
+            if (TryGetPrimitiveValue(out object val))
+            {
+                if (val is bool b)
+                {
+                    return b;
+                }
+            }
+
+            throw new InvalidOperationException($"Can't coerce to double from {this.Type._type.GetKindString()})");
+        }
+
+        public double AsDouble()
+        {
+            if (this is BlankValue)

```

Comment:
This is dangerous - blanks are different than 0.0, and it is ideal to mix those two concepts.

Reply:
Caller can always get blank if they care. But a common pattern is missing is default, some should make that easy.